base/number.jl

```diff
@@ -168,10 +168,23 @@ abs(x::Real) = ifelse(signbit(x), -x, x)
 
 Squared absolute value of `x`.
 
+For complex numbers this is faster than `abs(x)^2`.
```

Comment:
```suggestion
This can be faster than `abs(x)^2`, especially for complex
numbers where `abs(x)` requires a square root via [`hypot`](@ref).
```
It can also be faster for real numbers depending on the type, because
LLVM may not always be smart enough to eliminate the sign check. It can also be faster for other numeric types such as quaternions.